src/langcoder.cc

```diff
@@ -208,19 +205,19 @@ QMap< QString, GDLangCode > LangCoder::LANG_CODE_MAP = {
 
 QString LangCoder::decode( quint32 _code )
 {
-  if ( auto code = intToCode2( _code ); exists( code ) )
+  if ( auto code = intToCode2( _code ); code2Exists( code ) )
     return QString::fromStdString( LANG_CODE_MAP[ code ].lang );
 
   return {};
 }
-bool LangCoder::exists( const QString & _code )
+bool LangCoder::code2Exists( const QString & _code )
```

Comment:
where does `2` come from ?

Reply:
code2 is 2 char lang code
code3 is 3 char lang code

code2 is used in surrounding code, and it is the key of LANG_CODE_MAP

https://github.com/xiaoyifang/goldendict-ng/blob/6a91c6bde34c5d37f2c110a9a6c57b72cce3bb51/src/langcoder.cc#L20-L21

https://github.com/xiaoyifang/goldendict-ng/blob/6a91c6bde34c5d37f2c110a9a6c57b72cce3bb51/src/langcoder.hh#L34

https://github.com/xiaoyifang/goldendict-ng/blob/6a91c6bde34c5d37f2c110a9a6c57b72cce3bb51/src/langcoder.hh#L8-L11

Reply:
Maybe we should alias those types to make things clearer
```
using id = quint32
using code2 = QString
using code3 = QString
```

Reply:
I suggest that add an comment about the explaination of `code2`

Reply:
ok